### ISSUE TITLE
Sdk 46 dev client build error: Could not find: dso to load libhermes-executor-release.so result: 0

### ISSUE THREAD

**Issue by kastriotkastrati** [expo/expo#18825]
### Summary

I'm using expo managed workflow with hermes, upgrading from expo 45 to 46. 
I made a dev client build and it was successful, however, whenever I open the client it immediately crashes with this error.

```
Could not find: dso to load libhermes-executor-release.so
```

This same issues is resolvable on bare workflows by editing the gradle file according to this issue, but I found no resolution for managed workflow:
https://github.com/expo/expo/issues/18275

### Managed or bare workflow?

managed

### What platform(s) does this occur on?

Android

### Package versions

```
"expo-dev-client": "~1.2.1",
"expo": "~46.0.9",
"react-native": "0.69.4",
```

### Environment

System: OS Windows 11
Managed workflow

### Reproducible demo

Minimal dev client build with hermes, expo sdk46.

**Comment by almirfilho** [comment]
@kastriotkastrati did you manage to fix it? I'm having a similar issue after upgrading to SDK 47 (dev client also).

**Comment by kastriotkastrati** [comment]
@almirfilho hey yeah. Sorry the fix was anticlimactic so I just closed the issue. Although I had mentioned there that I ran in a managed workflow, this terminology is outdated. 

As I recall, some of the steps on the build process on expo servers were being skipped in my case since I was already providing the ios/ and android/ folders. This would have been totally okay if I had fully updated the ios/ and android/ folders but I had not done that. 

The way I solved this is by fully updating my ios/ and android/ folders with: `npx expo prebuild --clean`. The `--clean` flag is the important bit since this regenerates everything from the beginning.
